scala/wayfinder/src/main/scala/com/azavea/wayfinder/json/RoutingRequestParams.scala

```diff
@@ -0,0 +1,37 @@
+package com.azavea.wayfinder.json
+
+import argonaut._, Argonaut._
+
+import LatLngJson._
+
+object RouteParamsJson {
+
+  case class RouteParams(
+    wheels: Boolean,
+    challenge: String,
+    origin: LatLng,
+    destination: LatLng,
+    walkLength: Double,
+    options: String
+  )
+
+  implicit def RouteParamsCodecJson: CodecJson[RouteParams] =
+    CodecJson(
+      (p: RouteParams) =>
+        ("wheels" := p.wheels) ->:
+        ("challenge" := p.challenge) ->:
+        ("origin" := p.origin) ->:
+        ("destination" := p.destination) ->:
+        ("walkLength" := p.walkLength) ->:
+        ("options" := p.options) ->:
+        jEmptyObject,
+      c => for {
+        wheels <- (c --\ "wheels").as[Boolean]
+        challenge <- (c --\ "challenge").as[String]
+        origin <- (c --\ "origin").as[LatLng]
+        destination <- (c --\ "destination").as[LatLng]
+        walkLength <- (c --\ "walkLength").as[Double]
+        options <- (c --\ "options").as[String]
+      } yield RouteParams(wheels, challenge, origin, destination, walkLength, options))
+}
```

Comment:
IIRC it is amongst the faster parser implementations. It has a bunch of functionality of manipulating the JSON tree, which doesn't seem very useful at first glance. I'd like to see it used to find out if there are some cool/useful uses that emerge out of that. spray-json is definitely much more focused and much more commonly used. Really there is no reason not to play around with Arg here, especially if we're not using geotrellis vector stuff and it's geojson serializers.